providers/implementations/signature/sm2_sig.c

```diff
@@ -313,6 +313,7 @@ int sm2sig_digest_verify_final(void *vpsm2ctx, const unsigned char *sig,
 
     if (psm2ctx == NULL
         || psm2ctx->mdctx == NULL
+        || EVP_MD_get_size(psm2ctx->md) <= 0
```

Comment:
Others might disagree, but I would normally break this up into 2 if statements, with a  sz = EVP_MD_get_size(); between them.

Reply:
Yeah, do not call EVP_MD_get_size() twice.

Reply:
Thanks. I have add a new commit to solve the problem.